lib/matplotlib/axes/_axes.py

```diff
@@ -2274,10 +2274,20 @@ def bar(self, x, height, width=0.8, bottom=None, *, align="center",
         # left, bottom, width, height vectors
         if align == 'center':
             if orientation == 'vertical':
-                left = x - width / 2
+                try:
+                    left = x - width / 2
+                except TypeError as e:
+                    raise ValueError('the dtypes of parameters x '
```

Comment:
Should still be a TypeError? I don't see why you'd change the exception type tbh.
Also, sorry for the OCDness, but can you put the linebreaks at the same place for x and y? :)

Reply:
Yeah, I would have done that - @timhoffm suggest value error, but maybe mistakenly?  

As for OCD, admitting that you have a problem is the first step :wink:

Reply:
Changed to a TypeError - if @timhoffm has a reason for it not to be, he can comment.  Aligned the docstrings, though "height" is a longer word the "width", so there is only so much I can do there :wink: